File: examples/isogram/isogram.lua
---@param s string
---@return boolean
local function is_isogram(s)
    local seen = {}
    for c in s:lower():gmatch "%a" do
        if seen[c] then
            return false
        else
            seen[c] = true
        end
    end
    return true
end

return is_isogram

File: examples/isogram/isogram.rs
pub fn is_isogram(s: &str) -> bool {
    let mut seen = std::collections::HashSet::new();
    for c in s.to_lowercase().chars().filter(|c| c.is_alphabetic()) {
        if !seen.insert(c) {
            return false;
        }
    }
    true
}
